### PR TITLE
Add oidc docker example with keycloak #1566

### PR DESCRIPTION
**Pull request by AntonEliatra**
### Description
Adding oidc docker example with keycloak configuration to provide roles

### Issues Resolved
Part of issues on https://github.com/opensearch-project/documentation-website/issues/1566

### Version
all

### Checklist
- [x] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by scrawfor99 — APPROVED** [review]

File: _security/authentication-backends/openid-connect.md [hunk 1]
```diff
@@ -413,3 +413,20 @@ config:
...
+Following steps are necessary to get this up and running:
+
+1. Download and unzip the [example zip file.]({{site.url}}{{site.baseurl}}/assets/examples/oidc_example.zip)
+2. Update the `.env` file with a strong password for `admin` user.
+3. Substitute the `{IP}` placeholders in `config.yml` and `opensearch_dashboards.yml` with the IP of the local machine.
+4. Review the files:
+  - `docker-compose.yml` defines a single OpenSearch node, OpenSearch Dashboards and Keycloak server.
+  - `new-realm.json` specifies the details for the realm named `new`
+  - `config.yml` configures `basic_internal_auth_domain` and `oidc_auth_domain`
+  - `opensearch_dashboards.yml` configured to point to Keycloak for authentication
```
Comment (Naarcha-AWS) [inline _security/authentication-backends/openid-connect.md:430]:
```suggestion
  - `opensearch_dashboards.yml` should point to Keycloak for authentication. Make sure that the `opensearch_security.openid.connect_url` setting points to the URL of the realm.

File: _security/authentication-backends/openid-connect.md [hunk 2]
```diff
@@ -413,3 +413,20 @@ config:
...
+5. At the command line, run `docker-compose up`.
+6. Access OpenSearch Dashboards on `http://localhost:5601` and login with user `testuser` and password `testpasword` (which is configured in `new-realm.json`). Upon login the `testuser` receives backend role `admin` from Keycloak, which is mapped to `all_access` OpenSearch role. These backend roles can be managed using Keycloak `Administrative Console` at http://localhost:8080 using username `admin` and password `admin`.
```
Comment (Naarcha-AWS) [inline _security/authentication-backends/openid-connect.md:432]:
```suggestion
6. Access OpenSearch Dashboards on `http://localhost:5601` and login with user `testuser` and password `testpassword` configured in `new-realm.json`. After logging in the `testuser` receives backend role `admin` from Keycloak, which is mapped to the `all_access` OpenSearch role. These backend roles can be managed using the Keycloak Administrative Console at http://localhost:8080, using username `admin` and password `admin`.
```

**Review by Naarcha-AWS — CHANGES_REQUESTED** [review]

File: _security/authentication-backends/openid-connect.md [hunk 3]
```diff
@@ -413,3 +413,19 @@ config:
...
+3. Substitute the `{IP}` placeholders in `config.yml` and `opensearch_dashboards.yml` with the IP of the local machine.
+4. Review the following files:
+  - `docker-compose.yml` defines a single OpenSearch node, OpenSearch Dashboards and Keycloak server.
+  - `new-realm.json` specifies the details for the [realm](https://www.keycloak.org/docs/latest/server_admin/#core-concepts-and-terms). In this example, the realm is named `new`.
+  - `config.yml` configures `basic_internal_auth_domain` and `oidc_auth_domain`.
+  - `opensearch_dashboards.yml` should point to Keycloak for authentication. Make sure that the `opensearch_security.openid.connect_url` setting points to the URL of the realm.
+5. At the command line, run `docker-compose up`.
+6. Access OpenSearch Dashboards on `http://localhost:5601` and login with user `testuser` and password `testpassword` configured in `new-realm.json`. After logging in the `testuser` receives backend role `admin` from Keycloak, which is mapped to the `all_access` OpenSearch role. These backend roles can be managed using the Keycloak Administrative Console at http://localhost:8080, using username `admin` and password `admin`.
```
Comment (Naarcha-AWS) [inline _security/authentication-backends/openid-connect.md]:
```suggestion
6. Access OpenSearch Dashboards on `http://localhost:5601` and login with user `testuser` and password `testpassword` configured in the `new-realm.json` file. 

After logging in the `testuser` receives backend role `admin` from Keycloak, which is mapped to the `all_access` OpenSearch role. These backend roles can be managed using the Keycloak Administrative Console at http://localhost:8080, using username `admin` and password `admin`.
```

**Review by Naarcha-AWS — APPROVED** [review]

File: _security/authentication-backends/openid-connect.md [hunk 4]
```diff
@@ -413,3 +413,21 @@ config:
...
+## Docker example with Keycloak
+
+The following steps use Docker and [Keycloak IdP](https://www.keycloak.org/) to set up a basic authentication backend:
+
+
+1. Download and unzip the [example OpenID Connect zip file]({{site.url}}{{site.baseurl}}/assets/examples/oidc_example.zip)
+2. Update the `.env` file with a strong password for the `admin` user.
+3. Substitute the `{IP}` placeholders in `config.yml` and `opensearch_dashboards.yml` with the IP of the local machine.
+4. Review the following files:
+  - `docker-compose.yml` defines a single OpenSearch node, OpenSearch Dashboards and Keycloak server.
```
Comment (natebower) [inline _security/authentication-backends/openid-connect.md:426]:
```suggestion
  - `docker-compose.yml` defines a single OpenSearch node, OpenSearch Dashboards, and Keycloak server.
```

File: _security/authentication-backends/openid-connect.md [hunk 5]
```diff
@@ -413,3 +413,21 @@ config:
...
+  - `new-realm.json` specifies the details for the [realm](https://www.keycloak.org/docs/latest/server_admin/#core-concepts-and-terms). In this example, the realm is named `new`.
```
Comment (natebower) [inline _security/authentication-backends/openid-connect.md:427]:
```suggestion
  - `new-realm.json` specifies the details of the [realm](https://www.keycloak.org/docs/latest/server_admin/#core-concepts-and-terms). In this example, the realm is named `new`.
```

File: _security/authentication-backends/openid-connect.md [hunk 6]
```diff
@@ -413,3 +413,21 @@ config:
...
+  - `config.yml` configures `basic_internal_auth_domain` and `oidc_auth_domain`.
+  - `opensearch_dashboards.yml` should point to Keycloak for authentication. Make sure that the `opensearch_security.openid.connect_url` setting points to the URL of the realm.
+5. At the command line, run `docker-compose up`.
+6. Access OpenSearch Dashboards on `http://localhost:5601` and login with user `testuser` and password `testpassword` configured in the `new-realm.json` file.
```
Comment (natebower) [inline _security/authentication-backends/openid-connect.md:431]:
```suggestion
6. Access OpenSearch Dashboards at `http://localhost:5601` and log in with username `testuser` and password `testpassword` configured in the `new-realm.json` file. 
```

File: _security/authentication-backends/openid-connect.md [hunk 7]
```diff
@@ -413,3 +413,21 @@ config:
...
+3. Substitute the `{IP}` placeholders in `config.yml` and `opensearch_dashboards.yml` with the IP of the local machine.
+4. Review the following files:
+  - `docker-compose.yml` defines a single OpenSearch node, OpenSearch Dashboards and Keycloak server.
+  - `new-realm.json` specifies the details for the [realm](https://www.keycloak.org/docs/latest/server_admin/#core-concepts-and-terms). In this example, the realm is named `new`.
+  - `config.yml` configures `basic_internal_auth_domain` and `oidc_auth_domain`.
+  - `opensearch_dashboards.yml` should point to Keycloak for authentication. Make sure that the `opensearch_security.openid.connect_url` setting points to the URL of the realm.
+5. At the command line, run `docker-compose up`.
+6. Access OpenSearch Dashboards on `http://localhost:5601` and login with user `testuser` and password `testpassword` configured in the `new-realm.json` file. 
+
+After logging in the `testuser` receives backend role `admin` from Keycloak, which is mapped to the `all_access` OpenSearch role. These backend roles can be managed using the Keycloak Administrative Console at http://localhost:8080, using username `admin` and password `admin`.
```
Comment (natebower) [inline _security/authentication-backends/openid-connect.md:433]:
```suggestion
After logging in, the `testuser` receives the backend role `admin` from Keycloak, which is mapped to the `all_access` OpenSearch role. These backend roles can be managed using the Keycloak Administrative Console at http://localhost:8080, using username `admin` and password `admin`.
```

**Review by natebower — COMMENTED** [review]
@Naarcha-AWS @AntonEliatra Please see my changes and let me know if you have any questions. Thanks!

**Pull request merged**